src/main/java/com/synopsys/integration/detect/tool/signaturescanner/BlackDuckSignatureScannerTool.java

```diff
@@ -121,8 +127,15 @@ private ScanBatchRunner createScanBatchRunner(@Nullable BlackDuckServerConfig bl
             if (signatureScannerOptions.getUserProvidedScannerInstallUrl().isPresent()) {
                 logger.debug("Signature scanner will use the provided url to download/update the scanner.");
                 String providedUrl = signatureScannerOptions.getUserProvidedScannerInstallUrl().get();
+                HttpUrl baseUrl;
+                try {
+                    baseUrl = new HttpUrl(providedUrl);
+                } catch (IntegrationException e) {
+                    throw new DetectUserFriendlyException("User provided scanner install url could not be parsed: " + providedUrl, e, ExitCodeType.FAILURE_CONFIGURATION);
+                }
+                UserAgentItem solutionUserAgentItem = createSolutionUserAgentItem();
                 IntHttpClient restConnection = connectionFactory.createConnection(providedUrl, new SilentIntLogger()); //TODO: Should this be silent?
-                BlackDuckHttpClientWrapper fakeBlackDuckHttpClient = new BlackDuckHttpClientWrapper(restConnection);
+                BlackDuckHttpClientWrapper fakeBlackDuckHttpClient = new BlackDuckHttpClientWrapper(restConnection, baseUrl, solutionUserAgentItem);
```

Comment:
I like calling it fake here. We might want to name the class "Fake" as well, and throw a comment in there as to why we're doing it this way.